### PR TITLE
[flutter_tools] remove more mocks from cache_test

### PR DESCRIPTION
**Pull request by jonahwilliams**
Part of #71511.

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/cache_test.dart [hunk 1]
```diff
@@ -971,8 +968,33 @@ class FakeDownloadedArtifact extends CachedArtifact {
   Future<void> updateInner(ArtifactUpdater artifactUpdater, FileSystem fileSystem, OperatingSystemUtils operatingSystemUtils) async { }
 }
 
-class MockCachedArtifact extends Mock implements CachedArtifact {}
-class MockIosUsbArtifacts extends Mock implements IosUsbArtifacts {}
+class MockCachedArtifact extends Fake implements CachedArtifact {
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/cache_test.dart:971]: `FakeCachedArtifact`

**Comment by jmagman** [comment]
Ruh roh
```
  error • The name 'FakeCachedArtifact' is already defined • packages/flutter_tools/test/general.shard/cache_test.dart:971:7 • duplicate_definition

```

**Review by jmagman — APPROVED** [review]
Or you could `hide FakeCachedArtifact`.   Might want a comment for why it's `Secondary` since that's not obvious.

LGTM

**Comment by jonahwilliams** [comment]
but FakeCachedArtifact is in the same file as this one

**Comment by jmagman** [comment]
> but FakeCachedArtifact is in the same file as this one

Oh, I thought it was coming from `context.dart`.  I didn't check out the PR to verify.  Carry on.

**Pull request merged**